src/Microsoft.TestPlatform.ObjectModel/ITelemetryInitializer.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.ObjectModel;
+
+/// <summary>
+/// Interface for extensions that choose to send telemetry events
+/// </summary>
+public interface ITelemetryInitializer
+{
+    /// <summary>
+    /// Initializes telemetry reporter
+    /// </summary>
+    void Initialize(ITelemetryReporter telemetryReporter);
```

Comment:
I wonder if we should have an option object for future extensibility but maybe we can add to the `ITelemetryReporter` self if needed

Reply:
I think extending API for extensions is not issue anymore.